### PR TITLE
Feature/orb create datasets

### LINKED ISSUE #272: Dataset creation

### PR DESCRIPTION
**Pull request by lterranova**
Resolves #272

### CODE REVIEW

File: policies/api/http/endpoint_test.go [hunk 1]
```diff
@@ -40,11 +40,15 @@ visor:
...
-	contentType  = "application/json"
-	wrongID      = "28ea82e7-0224-4798-a848-899a75cdc650"
+	limit                    = 10
+	invalidToken             = "invalid"
+	maxNameSize              = 1024
+	contentType              = "application/json"
+	wrongID                  = "28ea82e7-0224-4798-a848-899a75cdc650"
+	invalidJson              = "{"
+	validDatasetJson         = "{\n    \"name\": \"my-dataset-json\",\n    \"agent_group_id\": \"8fd6d12d-6a26-5d85-dc35-f9ba8f4d93db\",\n    \"agent_policy_id\": \"86b7b412-1b7f-f5bc-c78b-f79087d6e49b\",\n    \"sink_id\": \"f5b2d342-211d-a9ab-1233-63199a3fc16f\"\n,\n    \"tags\": {\n        \"region\": \"eu\",\n        \"node_type\": \"dns\"\n    }}"
+	validDatasetYaml         = `{"name": "my-dataset-yaml", "agent_group_id": "8fd6d12d-6a26-5d85-dc35-f9ba8f4d93db", "agent_policy_id": "86b7b412-1b7f-f5bc-c78b-f79087d6e49b", "sink_id": "urn:uuid:f5b2d342-211d-a9ab-1233-63199a3fc16f", "tags": {"region": "eu", "node_type": "dns"}}`
```
Comment (dscabral) [inline policies/api/http/endpoint_test.go:50]: I'm not sure if we have a yaml format when create a dataset. I think we just use the policy, agent groups and sinks ids... Please check it.

File: policies/api/http/endpoint_test.go [hunk 2]
```diff
@@ -469,6 +473,81 @@ func TestPolicyRemoval(t *testing.T) {
...
+	createDataset(t, &cli, "my-dataset-conflict")
+
+	cases := map[string]struct {
+		req         string
+		contentType string
+		auth        string
+		status      int
+		location    string
+	}{
+		"add a valid yaml dataset": {
```
Comment (dscabral) [inline policies/api/http/endpoint_test.go:490]: What the difference between validDatasetYaml and validDatasetJson?

**Review by dscabral — CHANGES_REQUESTED** [review]

File: policies/api/http/responses.go [hunk 3]
```diff
@@ -99,9 +100,32 @@ type pageRes struct {
...
 type datasetRes struct {
-	ID      string `json:"id"`
-	Name    string `json:"name"`
-	created bool
+	ID           string         `json:"id"`
+	Name         string         `json:"name"`
+	Valid        bool           `json:"valid"`
+	AgentGroupID string         `json:"agent_group_id"`
+	PolicyID     string         `json:"policy_id"`
+	SinkID       []string       `json:"sink_ids"`
```
Comment (weyrick) [inline policies/api/http/responses.go:108]: We should stay consistent and call this `SinkIDs`

File: policies/policy_service.go [hunk 4]
```diff
@@ -62,6 +62,7 @@ func (s policiesService) AddDataset(ctx context.Context, token string, d Dataset
 	if err != nil {
 		return Dataset{}, errors.Wrap(ErrCreateDataset, err)
 	}
+	//TODO create a function to retrieve a dataset by id
```
Comment (weyrick) [inline policies/policy_service.go:65]: Is this part of a different PR?

**Review by weyrick — CHANGES_REQUESTED** [review]

File: policies/policy_service.go (hunk 4 above)
Comment (lterranova) [reply policies/policy_service.go]: I guess it came when merging the develop to fix the conflicts, sorry

**Review by weyrick — APPROVED** [review]

**Review by dscabral — APPROVED** [review]

**Pull request merged**